resources/ts/plan-details.ts

```diff
@@ -0,0 +1,199 @@
+$(document).on('click', '[id^="remove_button_"]', function() {
+    const id: string = $(this).attr('id') || '';
+    const key: string = id.split('_')[2];
+    const cardElement: JQuery<HTMLElement> = $(`#place_${key}`);
+
+    let arrivalTime: string | undefined;
+    if (cardElement.length > 0) {
+        arrivalTime = cardElement.attr('data-arrival-time');
+    }
+
+    const newContent: string = `
+        <div class="card-body row">
+            <div class="col-10">
+                <div> <i class="fa-regular fa-clock"></i> ${arrivalTime}</div>
+                <div class="h5">Please add Place</div>
+            </div>
+            <div class="col-2 d-flex align-items-center">
+                <button name='add_button' id='add_button_${key}' type="button" class="border-0 bg-white" data-bs-toggle="modal" data-bs-target="#add_place_modal">
+                    <i class="fa-solid fa-plus"></i>
+                </button>
+            </div>
+        </div>
+    `;
+    cardElement.empty().append(newContent);
+});
+
+
+$(document).on('click', '[id^="add_button_"]', function() {
+    const buttonId: string = $(this).attr('id') || '';
+    const key: string = buttonId.split('_')[2];
+    const id: string = 'place_' + key;
+
+    sessionStorage.setItem('add_target_place_id', id);
+});
+
+
+$(document).on('click', '[id="add_place_button"]', function() {
+
+    const id = sessionStorage.getItem('add_target_place_id');
```

Comment:
If you are you are using a TypeScript, I would be more strict to this. please add what type is the id. 
e.g id: string = sessionStorage.getItem('add_target_place_id'); Whatever the type of id is.